test/functional/test_framework/chacha20.py

```diff
@@ -55,6 +57,34 @@ def chacha20_block(key, nonce, cnt):
     # Produce byte output
     return b''.join(state[i].to_bytes(4, 'little') for i in range(16))
 
+class FSChaCha20:
+    """Rekeying wrapper stream cipher around ChaCha20."""
+    def __init__(self, initial_key, rekey_interval=REKEY_INTERVAL):
+        self.key = initial_key
```

Comment:
small nit: not sure if it's really worth to change it, but in Python member variables and methods that shouldn't be accessed from the outside are usually prefixed with an underscore (`_`) (see e.g.: https://peps.python.org/pep-0008/#method-names-and-instance-variables). But that's just a naming convention anyway and the "private" usage is not enforced by the language.